### PR TITLE
Add dynamic OrderBy extensions for batched queries

### PR DESCRIPTION
**Pull request by smbecker**

### CODE REVIEW

**Review by mysticmind — COMMENTED** [review]
@smbecker could you please add a unit test as well?

**Comment by smbecker** [comment]
> @smbecker could you please add a unit test as well?

I duplicated the existing dynamic order-by tests leveraging a batched query

**Review by mysticmind — APPROVED** [review]
Looks good to me 👍

File: src/Marten/QueryableExtensions.cs [hunk 1]
```diff
@@ -492,43 +555,74 @@ public static IOrderedQueryable<T> ThenBy<T>(this IOrderedQueryable<T> queryable
...
+            return ApplyOrder(queryable, property, "ThenByDescending");
         }
 
         private static IOrderedQueryable<T> ApplyOrder<T>(
             IQueryable<T> queryable,
             string property,
             string methodName)
         {
+            var lambda = CompileOrderBy<T>(property, out var targetType);
+            var result = typeof(Queryable).GetMethods().Single(
```
Comment (oskardudycz) [inline src/Marten/QueryableExtensions.cs:579]: **COMMENT:** Not for this PR, but some memoisation will be good to add here.

**Review by oskardudycz — APPROVED** [review]

File: src/Marten/QueryableExtensions.cs (hunk 1 above)
Comment (mysticmind) [reply src/Marten/QueryableExtensions.cs:579]: We will add a TODO comment for this, will merge the PR.

**Pull request merged**